rockusb/src/libusb.rs

```diff
@@ -462,3 +467,21 @@ where
         Ok(self.offset)
     }
 }
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum BootMode {
+    Maskrom = 0,
+    Loader = 1,
+}
+
+impl BootMode {
+    pub fn from_usb_version(version: &rusb::Version) -> Result<BootMode> {
+        let boot_mode = match version.sub_minor() & 0x1 {
+            0 => BootMode::Maskrom,
+            1 => BootMode::Loader,
+            _ => return Err(Error::UsbError(rusb::Error::NotFound)),
```

Comment:
It looks like the other match arms are exhaustive, so the `_ => ` case shouldn't be reachable.

Does the rustc compiler *require* this be present? If so, `_ => unreachable!(),` is probably the way to go here.

Reply:
fwiw yes it will require it  to match all cases; the compiler doesn't understand that `& 0x1`  will limit the range to 0/1 -- so a default case will be needed.  potentially nicer then unreachable() would be to simply use `_ => BootMode::Loader` so you wouldn't need a pointless third branch :)